src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLSPOptionsMonitor.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorLSPOptionsMonitor : IOptionsMonitor<RazorLSPOptions>
+    {
+        private readonly ILanguageServer _server;
+        private readonly RazorConfigurationService _configurationService;
+        private readonly IOptionsMonitorCache<RazorLSPOptions> _cache;
+        internal event Action<RazorLSPOptions, string> _onChange;
+        private RazorLSPOptions _currentValue;
+
+        public RazorLSPOptionsMonitor(ILanguageServer server, RazorConfigurationService configurationService, IOptionsMonitorCache<RazorLSPOptions> cache)
+        {
+            if (server is null)
+            {
+                throw new ArgumentNullException(nameof(server));
+            }
+
+            if (configurationService is null)
+            {
+                throw new ArgumentNullException(nameof(configurationService));
+            }
+
+            if (cache is null)
+            {
+                throw new ArgumentNullException(nameof(cache));
+            }
+
+            _server = server;
+            _configurationService = configurationService;
+            _cache = cache;
+            Initialize();
+        }
+
+        public RazorLSPOptions CurrentValue => Get(Options.DefaultName);
+
+        public RazorLSPOptions Get(string name)
+        {
+            name = name ?? Options.DefaultName;
+            return _cache.GetOrAdd(name, () => _currentValue);
+        }
+
+        public IDisposable OnChange(Action<RazorLSPOptions, string> listener)
+        {
+            var disposable = new ChangeTrackerDisposable(this, listener);
+            _onChange += disposable.OnChange;
+            return disposable;
+        }
+
+        public async Task UpdateAsync()
+        {
+            var latestOptions = await _configurationService.GetLatestOptions();
+            if (latestOptions != null)
```

Comment:
What happens if they've actively removed their razor config somehow? Is such a thing even possible?

Reply:
Could you clarify a bit more. Who is "they" in this case?

Reply:
"They" in this case is either the end-user or client. I'm imagining something like a `.vscode/razor.config` file that gets deleted, or someone going into their VSCode config settings and deleting all of the ones for Razor.

It seems like the primary case when `GetLatestOptions` would return null is if result is null or `result.length==0`, but those sound like either a transport/protocol error or a client not abiding by the LSP spec, since our request will always have a single ConfigurationItem. In that case oughtn't we throw an error rather passing null down for later use?

Reply:
I took the approach of being a little forgiving because we don't want to throw every time something goes wrong with with JsonRPC etc. I'm still logging a warning if something super unexpected happens.

Reply:
I just realized I didn't add any warning for the result == 0 case. Added one.